crates/burn-tensor/src/tensor/ops/tensor.rs

```diff
@@ -1046,17 +1046,49 @@ pub trait FloatTensorOps<B: Backend> {
     /// A tensor with the same shape as `tensor` with error function values.
     fn float_erf<const D: usize>(tensor: FloatTensor<B, D>) -> FloatTensor<B, D>;
 
-    /// Catcatenates tensors along a dimension.
+    /// Concatenates tensors along a dimension.
     ///
     /// # Arguments
     ///
-    /// * `tensors` - The tensors to catcatenate.
-    /// * `dim` - The dimension along which to catcatenate.
+    /// * `tensors` - The tensors to concatenate.
+    /// * `dim` - The dimension along which to concatenate.
     ///
     /// # Returns
     ///
-    /// A tensor with the catcatenated tensors along `dim`.
-    fn float_cat<const D: usize>(tensors: Vec<FloatTensor<B, D>>, dim: usize) -> FloatTensor<B, D>;
+    /// A tensor with the concatenated tensors along `dim`.
+    fn float_cat<const D: usize>(tensors: Vec<FloatTensor<B, D>>, dim: usize) -> FloatTensor<B, D> {
+        let first_tensor = tensors.first().expect("Tensors should not be empty");
+        let mut shape = B::float_shape(first_tensor);
+        let device = &B::float_device(first_tensor);
+
+        let output_dim_length: usize = tensors
+            .iter()
+            .map(|tensor: &FloatTensor<B, D>| B::float_shape(tensor).dims[dim])
+            .sum();
+        shape.dims[dim] = output_dim_length;
+
+        let mut tensor_output = B::float_empty(shape.clone(), device);
+
+        let mut i = 0;
+        let indices_select_all = [0; D].map(|_| {
+            let start = 0;
+            let end = shape.dims[i];
+            i += 1;
+            start..end
+        });
+
+        let mut output_index = 0;
+        for tensor in tensors {
+            let mut indices = indices_select_all.clone();
+            let tensor_dim_length = B::float_shape(&tensor).dims[dim];
+            indices[dim] = output_index..tensor_dim_length;
+            output_index += tensor_dim_length;
+
+            tensor_output = B::float_slice_assign(tensor_output, indices, tensor)
+        }
+
+        tensor_output
```

Comment:
I think it would be better to implement this function in a specific module that work with the `Tensor` struct instead of the backend API. Here we could simply:

```rust
fn float_cat<const D: usize>(tensors: Vec<FloatTensor<B, D>>, dim: usize) -> FloatTensor<B, D> {

Reply:
I tried, see #1473